hello-world/test/main.test.ts

```diff
@@ -1,31 +1,39 @@
-import { expect } from 'chai';
-import { Wallet, Provider, Contract } from 'zksync-web3';
-import * as hre from 'hardhat';
-import { Deployer } from '@matterlabs/hardhat-zksync-deploy';
+import { expect } from "chai";
+import { deploy } from "./utils/utils";
 
-const RICH_WALLET_PK =
-  '0x7726827caac94a7f9e1b160f7ea819f172f7b6f9d2a97f992c38edeab82d4110';
+describe("Greeter", function () {
+  let contract: any;
+  let result: string;
 
-async function deployGreeter(deployer: Deployer): Promise<Contract> {
-  const artifact = await deployer.loadArtifact('Greeter');
-  return await deployer.deploy(artifact, ['Hi']);
-}
+  beforeEach(async function () {
+    contract = await deploy();
+  });
 
-describe('Greeter', function () {
-  it("Should return the new greeting once it's changed", async function () {
-    const provider = Provider.getDefaultProvider();
+  it("Should be deployed and have address", async function () {
+    result = typeof (await contract.address);
+
+    expect(result).to.be.a("string");
+  });
+
+  it("Should be deployed and have tx hash", async function () {
+    result = await contract.deployTransaction.hash;
 
-    const wallet = new Wallet(RICH_WALLET_PK, provider);
-    const deployer = new Deployer(hre, wallet);
+    expect(result).to.be.a("string");
+  });
+
+  it("Should return 'Hi' as expected message", async function () {
+    result = await contract.greet();
 
-    const greeter = await deployGreeter(deployer);
+    expect(result).to.eq("Hi");
+  });
+
+  it("Should return the new greeting once it's changed", async function () {
+    const setGreetingTx = await contract.setGreeting("Hola, mundo!");
 
-    expect(await greeter.greet()).to.eq('Hi');
+    await setGreetingTx.wait(1); // wait until the transaction is mined
 
-    const setGreetingTx = await greeter.setGreeting('Hola, mundo!');
-    // wait until the transaction is mined
-    await setGreetingTx.wait();
+    result = await contract.greet();
 
-    expect(await greeter.greet()).to.equal('Hola, mundo!');
+    expect(result).to.equal("Hola, mundo!");
   });
 });
```

Comment:
I'd try adding some failing tests too, like passing wrong param types for example. Important to demonstrate that testing also involves testing unexpected scenarios